core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -997,9 +997,8 @@ public AttachDetachStatus detachPartition(long timestamp) {
         }
 
         // To detach the partition, squash it into single folder if required
-        squashSplitPartitions(timestamp, txWriter.ceilPartitionTimestamp(timestamp), 1);
+        squashPartitionForce(partitionIndex);
 
-        partitionIndex = txWriter.getPartitionIndex(timestamp);
         // Get next partition, should exist, it's not the last partition
         if (txWriter.getLogicalPartitionTimestamp(txWriter.getPartitionTimestampByIndex(partitionIndex + 1)) == timestamp) {
             // Could not squash to single partition before detaching because of active table readers.
```

Comment:
Is this check still relevant? Since the squashing always succeeds, we could turn it into an assertion maybe?

Reply:
marking this one as a nit since it's not a blocker